addons/mail/models/mail_thread.py

```diff
@@ -2617,6 +2612,15 @@ def _notify_classify_recipients(self, recipient_data, model_name, msg_vals=None)
             for group_name, group_func, group_data in groups:
                 if group_func(recipient):
                     group_data['recipients'].append(recipient['id'])
+                    group_button_access = group_data.setdefault('button_access', {})
```

Comment:
the setdefault doesn't seem necessary since we already did it in the "`# fill group_data with default_values if they are not complete`" loop
```suggestion
                    group_button_access = group_data['button_access']
```